src/components/Attachments/AttachmentCarousel/extractAttachmentsFromReport.js

```diff
@@ -15,10 +15,17 @@ import CONST from '@src/CONST';
 function extractAttachmentsFromReport(parentReportAction, reportActions) {
     const actions = [parentReportAction, ...ReportActionsUtils.getSortedReportActions(_.values(reportActions))];
     const attachments = [];
+    const uniqueSources = new Set();
```

Comment:
let's say some images are posted in a chat:

- X
- A
- B
- C
- A
- D
- A

you click on the last instance of `A`, then press `<` on the carousel. What happens? do you see X, C, or D?

I'd suggest D - if an attachment has been posted multiple times then it's position the the attachment carousel matches the _last_ time that it was posted.

Thoughts?

Reply:
I appreciate the concept, and I believe we can enhance user experience with a slight modification. When a user selects a duplicate, it would be ideal to treat it as if they had selected the original, automatically disregarding the other duplicates. This approach would maintain intuitive navigation.

Currently, our system does not distinguish which instance of `A` a user has selected. As a result, the user is directed to the first occurrence of `A`, which is located near the `X`, by default.

I am considering updating the logic to target the last instance of `A` instead. However, this solution merely shifts the problem. For instance, if a user selects the `A` adjacent to `X`, they would naturally expect to either return to `X` or advance to `B`.

The `ImageRenderer` provides some context by indicating which `ReportAction` has been selected. However, there is a discrepancy with the node index information when compared to the HTML parser used here. Specifically, the indexes for images differ because the HTML of messages isn't segmented in the same manner, primarily due to how text not enclosed by tags is processed.

I propose that we attempt to implement a straightforward solution for now and consider addressing this issue more comprehensively in a separate task.

Reply:
sure, let's just document the expected behavior resulting from the simplest solution in a code comment 👍🏼